ConfigSpace/hyperparameters.pyx

```diff
@@ -1483,7 +2265,8 @@ cdef class CategoricalHyperparameter(Hyperparameter):
         repr_str.write("}")
         repr_str.write(", Default: ")
         repr_str.write(str(self.default_value))
-        if self.probabilities is not None:
+        # if the probability distribution is not uniform, write out the probabilities
+        if not np.all(self.probabilities == self.probabilities[0]):
```

Comment:
If `weights` is `None` during construction, this will fail as `self.probabilties` is `None`

Reply:
Never mind, this was fixed